### PR TITLE
Server changes added locally

### PR DESCRIPTION
**Pull request by Georgehe4**
Ready for review. Awaiting server side changes.

### CODE REVIEW

**Comment by lukechurch** [comment]
@Georgehe4 Let's co-ordinate by email when you'd like me to do a server push

**Comment by Georgehe4** [comment]
@lukechurch Sounds good.

**Comment by Georgehe4** [comment]
@devoncarew @lukechurch 
This is working on test servers at http://dev.dartproject-1000.appspot.com/. Note to manually change any URLs to the dev url if you are trying to use new functionality.
Open the URL in Firefox/Safari to work around certificate issues, or manually accept the certificate.
Eg: https://dev.dartproject-1000.appspot.com/?source=44489500-79c9-43cf-8b8a-41acfff316e3

**Comment by Georgehe4** [comment]
#623 #616

File: lib/playground.dart [hunk 1]
```diff
@@ -190,8 +191,10 @@ class Playground implements GistContainer, GistController {
         isLegalGistId(url.queryParameters['id'])) {
       _showGist(url.queryParameters['id']);
     } else if (url.hasQuery && url.queryParameters['export'] != null) {
-      UuidContainer requestId = new UuidContainer()..uuid=url.queryParameters['export'];
-      Future<PadSaveObject> exportPad = dartSupportServices.pullExportContent(requestId);
+      UuidContainer requestId = new UuidContainer()
+        ..uuid = url.queryParameters['export'];
+      Future<PadSaveObject> exportPad =
+          dartSupportServices.pullExportContent(requestId);
       await exportPad.then((pad) {
```
Comment (devoncarew) [inline lib/playground.dart]: this await isn't needed; we're not waiting for the return value

File: lib/playground.dart [hunk 2]
```diff
@@ -199,6 +202,15 @@ class Playground implements GistContainer, GistController {
         blankGist.getFile('styles.css').content = pad.css;
         editableGist.setBackingGist(blankGist);
       });
+    } else if (url.hasQuery && url.queryParameters['source'] != null) {
+      Future<UuidContainer> futureGistId = dartSupportServices
```
Comment (devoncarew) [inline lib/playground.dart]: ditto; this await isn't needed; we're not waiting for the return value

File: lib/playground.dart [hunk 3]
```diff
@@ -199,6 +202,15 @@ class Playground implements GistContainer, GistController {
...
+          .retrieveGist(id: url.queryParameters['source']);
+      await futureGistId.then((UuidContainer gistId) {
```
Comment (devoncarew) [inline lib/playground.dart]: this await isn't needed; we're not waiting for the return value

Comment (devoncarew) [inline lib/playground.dart]: And, generally, I'm not sure why this method is declared to return a Future when we not returning values from it

File: lib/playground.dart [hunk 4]
```diff
@@ -910,8 +928,8 @@ class GistFileProperty implements Property {
   }
 
   Stream get onChanged => file.onChanged.map((value) {
-    return value;
-  });
+        return value;
+      });
```
Comment (devoncarew) [inline lib/playground.dart]: Some indent issues; please format the file before you merge the PR.

**Comment by devoncarew** [comment]
lgtm! Let's merge it once the server has the new API.

**Comment by devoncarew** [comment]
From @lukechurch, the server is serving the new api.

**Comment by Georgehe4** [comment]
Note: Sauce disabled due to Travis CI build failure via 
https://travis-ci.org/dart-lang/dart-pad/builds/76329864

**Comment by Georgehe4** [comment]
Merging.

File: lib/playground.dart (hunk 1 above)
Comment (Georgehe4) [inline lib/playground.dart]: Note: Still kept in order to ensure up-to-date analysis

**Pull request merged**